test/runtests.jl

```diff
@@ -10,6 +10,58 @@ Base.size(x::TestArray) = size(x.x)
 Base.getindex(x::TestArray, i) = x.x[i]
 DataAPI.levels(x::TestArray) = reverse(DataAPI.levels(x.x))
 
+struct TestMeta
+    table::Dict
+    col::Dict
+
+    TestMeta() = new(Dict(), Dict())
+end
+
+function DataAPI.metadata(x::TestMeta, key; full::Bool=false)
+    return full ? x.table[key] : x.table[key][1]
+end
+
+function DataAPI.metadata(x::TestMeta, key, default; full::Bool=false)
+    haskey(x.table, key) && return DataAPI.metadata(x, key, full=full)
+    full ? (default, :none) : default
+end
+
+DataAPI.metadatakeys(x::TestMeta) = keys(x.table)
+
+function DataAPI.metadata!(x::TestMeta, key, value; style)
+    x.table[key] = (value, style)
+    return x
+end
+
+function DataAPI.colmetadata(x::TestMeta, col, key; full::Bool=false)
+    return full ? x.col[col][key] : x.col[col][key][1]
+end
+
+function DataAPI.colmetadata(x::TestMeta, col, key, default; full::Bool=false)
+    haskey(x.col, col) && haskey(x.col[col], key) && return DataAPI.colmetadata(x, col, key, full=full)
+    full ? (default, :none) : default
+end
+
+function DataAPI.colmetadatakeys(x::TestMeta, col)
+    haskey(x.col, col) && return keys(x.col[col])
+    return ()
+end
+
+function DataAPI.colmetadatakeys(x::TestMeta)
+    isempty(x.col) && return ()
+    return Any[col => keys(x.col[col]) for col in keys(x.col)]
```

Comment:
Not a big deal for this test, but wouldn't it be better in general to return a generator?
```suggestion
    return (col => keys(x.col[col]) for col in keys(x.col))
```